internal/plugins/ansible/v1/init.go

```diff
@@ -85,95 +95,81 @@ Optionally creates a new API, using the same flags as "create api"
-	p.apiSubc.config = p.config
+	fs.StringVar(&p.options.Group, "group", "", "resource Group")
+	fs.StringVar(&p.options.Version, "version", "", "resource Version")
+	fs.StringVar(&p.options.Kind, "kind", "", "resource Kind")
+	p.apiSubcommand.BindFlags(fs)
 }
 
-// createOptions with defaults set, for comparison in case GVK/chart inputs were set.
-var emptyCreateOptions = createOptions{CRDVersion: "v1"}
+func (p *initSubcommand) InjectConfig(c config.Config) error {
+	p.config = c
 
-func (p *initSubcommand) Run() error {
-	// Set values in the config
-	if err := p.config.SetProjectName(p.projectName); err != nil {
-		return err
-	}
 	if err := p.config.SetDomain(p.domain); err != nil {
 		return err
 	}
 
-	// Check if the project name is a valid k8s namespace (DNS 1123 label).
-	if p.config.GetProjectName() == "" {
+	// Assign a default project name
+	if p.projectName == "" {
 		dir, err := os.Getwd()
 		if err != nil {
 			return fmt.Errorf("error getting current directory: %v", err)
 		}
-
-		if err := p.config.SetProjectName(strings.ToLower(filepath.Base(dir))); err != nil {
-			return err
-		}
-	}
-
-	if err := cmdutil.Run(p); err != nil {
-		return err
-	}
-
-	// Run SDK phase 2 plugins.
-	if err := p.runPhase2(); err != nil {
-		return err
-	}
-
-	// If API creation is configured, run the 'create api' subcommand.
-	if p.apiSubc.options != emptyCreateOptions {
-		p.apiSubc.options.Domain = p.domain
-		if err := p.apiSubc.Run(); err != nil {
-			return err
-		}
+		p.projectName = strings.ToLower(filepath.Base(dir))
 	}
-
-	return nil
-}
-
-// SDK phase 2 plugins.
-func (p *initSubcommand) runPhase2() error {
-	if err := manifestsv2.RunInit(p.config); err != nil {
-		return err
+	// Check if the project name is a valid k8s namespace (DNS 1123 label).
+	if err := validation.IsDNS1123Label(p.projectName); err != nil {
+		return fmt.Errorf("project name (%s) is invalid: %v", p.projectName, err)
 	}
-	if err := scorecardv2.RunInit(p.config); err != nil {
+	if err := p.config.SetProjectName(p.projectName); err != nil {
 		return err
 	}
 
 	return nil
 }
 
-func (p *initSubcommand) Validate() error {
-	if err := validation.IsDNS1123Label(p.config.GetProjectName()); err != nil {
-		return fmt.Errorf("project name (%s) is invalid: %v", p.config.GetProjectName(), err)
-	}
-
-	return nil
-}
-
-func (p *initSubcommand) GetScaffolder() (cmdutil.Scaffolder, error) {
-	return scaffolds.NewInitScaffolder(p.config), nil
+func (p *initSubcommand) Scaffold(fs afero.Fs) error {
+	scaffolder := scaffolds.NewInitScaffolder(p.config)
+	scaffolder.InjectFS(fs)
+	return scaffolder.Scaffold()
 }
 
 func (p *initSubcommand) PostScaffold() error {
-	if p.apiSubc.options == emptyCreateOptions {
+	doAPI := p.options.Group != "" || p.options.Version != "" || p.options.Kind != ""
+	if !doAPI {
 		fmt.Printf("Next: define a resource with:\n$ %s create api\n", p.commandName)
+	} else {
+		args := []string{"create", "api"}
+		if p.options.Group != defaultGroup {
```

Comment:
If we define that the default group should be an example then it will not work. 
So, IMO the if need to be != "" and also we need to trim the spaces

Reply:
It is the other way round. We want to detect when the user did not provide any value for this flag. so we need to use the exact same value as the one provided in `BindFlags`. The concept here is, if the user did not provide the flag, we call `create api` without the flag.

Reply:
- Whate happens if I am as a user pass via the flag the string "       " in this implementation? 
- What happens if I as a contributor change the default value in the const defined with the goal of define default values always? Will that work? I do not think so. Because of this, I am not sure if I agree to much in creating the default const instead of use "" in the BindFlags as it was before. It can let the contributor misunderstand that.

Reply:
> * Whate happens if I am as a user pass via the flag(s) the string `"       "` in this implementation? We will try to create the api. Am I right? Would we want that?

But the validation will not pass because the "   " group is not a valid one, this is, this is left to the normal group flag validation. Here we are just saying, whatever the user gave to `operator-sdk init` regarding the api, pass it to a `operator-sdk create api` call. The create api call will validate it normally.

> * What happens if I as a contributor change the default value in the const defined with the goal of define default values always? Will that work? I do not think so. Because of this, I am not sure if I agree to much in creating the default const instead of use "" in the BindFlags as it was before. It can let the contributor misunderstand that. I'd prefer we keep the code with the default const == ""

Same as above, this is controlled and validated by the cretae api call, here we are just saying that any of them which was provided is passed, if they were not provided (i.e., they have the default value), then we do not add this field, and the default value for the create api call will be used.

Reply:
Why we need many const all with "" value and which never can be changed?

Reply:
They can have any value, the value itself doesn't matter. What matters is that both the default value assigned to the flag and the value checked here are the same constant. And then, why not just hard-code `""`? Because this allows to provide better help. Imagine for example that in a future the default version is set to `v1`. We could chnage the consta here to `v1` and it will still work perfectly, but the help message would show that the default value for this flag is `v1`.

Reply:
if I we change there to v1 then, it means that the default value would not be used and this is the point.

Reply:
However, since it is a nit and not a blocker I am closing this one. We can always change the followup.